domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/nocconfig/NoCConfigServiceImplTest.java

```diff
@@ -0,0 +1,113 @@
+    }
+
+    @Test
+    @DisplayName("Should get Noc Config entities for the passed case type references")
+    void shouldGetNoCConfigsForValidJurisdictions() {
+        List<NoCConfigEntity> noCConfigEntities = Lists.newArrayList();
+        noCConfigEntities.add(createNoCConfigEntity(true, true));
+        noCConfigEntities.add(createNoCConfigEntity(false, false));
+        doReturn(noCConfigEntities).when(nocConfigRepository).findAllByCaseTypeReferences(anyList());
+        List<String> references = Lists.newArrayList("Test", "Divorce");
+        List<NoCConfigEntity> configEntities = classUnderTest.getAll(references);
+        Assert.assertEquals(2, configEntities.size());
+    }
+
+    @Test
+    @DisplayName("Should get empty noc config list for the passed empty case references")
+    void shouldGetEmptyNoCConfigsListForEmptyJurisdictions() {
+        List<NoCConfigEntity> noCConfigEntities = Lists.newArrayList();
+        doReturn(noCConfigEntities).when(nocConfigRepository).findAllByCaseTypeReferences(anyList());
+        List<NoCConfigEntity> nocConfigsReturned = classUnderTest.getAll(Lists.newArrayList());
+        Assert.assertEquals(0, nocConfigsReturned.size());
+    }
+
+    @Test
+    @DisplayName("Should get empty noc config list for the passed empty case references")
+    void shouldSaveNoCConfigEntity() {
+        NoCConfigEntity noCConfigEntity = mock(NoCConfigEntity.class);
+        CaseTypeEntity caseTypeEntity = mock(CaseTypeEntity.class);
+        doReturn(caseTypeEntity).when(noCConfigEntity).getCaseType();
+        doReturn("TEST").when(caseTypeEntity).getReference();
+        doReturn(null).when(nocConfigRepository).findByCaseTypeReference(anyString());
+        classUnderTest.save(noCConfigEntity);
+        verify(nocConfigRepository, times(1)).save(eq(noCConfigEntity));
+    }
+
+    @Test
+    @DisplayName("Should get empty noc config list for the passed empty case references")
+    void shouldSaveCopyNoCConfigEntity() {
+        NoCConfigEntity configEntity = mock(NoCConfigEntity.class);
+        CaseTypeEntity caseTypeEntity = mock(CaseTypeEntity.class);
+        doReturn(caseTypeEntity).when(configEntity).getCaseType();
+        doReturn("TEST").when(caseTypeEntity).getReference();
+        NoCConfigEntity configEntityDB = mock(NoCConfigEntity.class);
+        doReturn(configEntityDB).when(nocConfigRepository).findByCaseTypeReference(anyString());
+        classUnderTest.save(configEntity);
+        verify(configEntityDB, times(1)).copy(eq(configEntity));
+        verify(nocConfigRepository, times(1)).save(eq(configEntityDB));
+    }
+
+    @Test
+    void shouldAttemptToDeleteNoCConfigEntriesWhenCaseTypeReferenceIsProvided() {
+        String reference = "TEST";
+
+        classUnderTest.deleteCaseTypeNocConfig(reference);
+
+        verify(nocConfigRepository, times(1)).deleteByCaseTypeReference(eq(reference));
+    }
+
+    @Test
+    void shouldNotAttemptToDeleteNoCCofigsWhenNoCaseReferenceIsProvided() {
```

Comment:
Maybe we could add @DisplayName for consistency.

Reply:
done